yank.c

```diff
@@ -316,18 +316,19 @@ tgetc(void)
 		{ "\033[D",	KEY_LEFT },
 		{ NULL,		0 },
 	};
-	char	buf[3];
+	char	buf[4];
 	ssize_t	n;
 	int	i;
 
-	n = read(tty.rfd, buf, sizeof(buf));
+	n = read(tty.rfd, buf, sizeof(buf) - 1);
+	buf[n] = 0;
```

Comment:
Caution here, `n` can be negative. Put this after `return KEY_TERM`. Also favor `'\0'` over `0`.

Reply:
Ah sorry I overlooked that. Fixed.